File: qqbot-runtime/src/main/java/com/mieai/qqbot/runtime/configuration/ConfigurationValidation.java
package com.mieai.qqbot.runtime.configuration;

import java.util.Objects;

final class ConfigurationValidation {
    private ConfigurationValidation() {}

    static String displayName(String value) {
        Objects.requireNonNull(value, "displayName must not be null");
        if (value.isBlank()) {
            throw new IllegalArgumentException("displayName must not be blank");
        }
        if (!value.equals(value.strip())) {
            throw new IllegalArgumentException("displayName must not have surrounding whitespace");
        }
        if (value.codePoints().anyMatch(Character::isISOControl)) {
            throw new IllegalArgumentException("displayName must not contain control characters");
        }
        return value;
    }
}

File: qqbot-runtime/src/main/kotlin/com/mieai/qqbot/runtime/configuration/ConfigurationValidation.kt
package com.mieai.qqbot.runtime.configuration

internal object ConfigurationValidation {
    fun displayName(value: String): String {
        require(value.isNotBlank()) { "displayName must not be blank" }
        require(value == value.trim()) { "displayName must not have surrounding whitespace" }
        require(value.codePoints().noneMatch(Character::isISOControl)) { "displayName must not contain control characters" }
        return value
    }
}
